File: 144.py
class Solution:
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        res = []
        if root is None:
            return res
        st = [root]
        while len(st) != 0:
            node = st.pop()
            res.append(node.val)
            if node.right is not None:
                st.append(node.right)
            if node.left is not None:
                st.append(node.left)
        return res

File: sol/144.cpp
struct Solution {
    vector<int> preorderTraversal(TreeNode* root) {
        vector<int> res;
        if (root == nullptr){
            return res;
        }
        vector<TreeNode*> st;
        st.push_back(root);
        while (!st.empty()){
            TreeNode *node = st.back();
            st.pop_back();
            res.push_back(node->val);
            if (node->right != nullptr){
                st.push_back(node->right);
            }
            if (node->left != nullptr){
                st.push_back(node->left);
            }
        }
        return res;
    }
};
